packages/datadog_flutter_plugin/ios/Classes/DatadogRumPlugin.swift

```diff
@@ -260,6 +264,258 @@ public class DatadogRumPlugin: NSObject, FlutterPlugin {
             result(FlutterMethodNotImplemented)
         }
     }
+
+    func callEventMapper<T>(mapperName: String, event: T, encodedEvent: [String: Any?], completion: ([String: Any?]?) -> T?) -> T? {
+        guard let methodChannel = DatadogRumPlugin.methodChannel else {
+            return event
+        }
+
+        var encodedResult: [String: Any?]? = encodedEvent
+        let semaphore = DispatchSemaphore(value: 0)
+
+        methodChannel.invokeMethod(mapperName, arguments: ["event": encodedEvent]) { result in
+            if result == nil {
+                encodedResult = nil
+            } else if let result = result as? [String: Any] {
+                encodedResult = result
+            }
+
+            semaphore.signal()
+        }
+
+        if semaphore.wait(timeout: .now() + DispatchTimeInterval.milliseconds(250)) == .timedOut {
+            Datadog._internal.telemetry.debug(id: "event_mapper_timeout", message: "\(mapperName) timed out.")
```

Comment:
question/ Do you foresee running into time outs often? If so, is there more context we could add to understand it better?

Reply:
I'm hoping we won't see it very often.  My own timing suggests that this should take very little time.

The one time we might see it would be if someone is stalling the main Dart thread. There's an improvement I'd like to make to the mappers to run in their own lightweight isolate, but that will require some extra work down the road.  This will tell me if I need to prioritize that.

Reply:
Did a bit more timing that suggests I might want to increase this for debug builds to 500ms.

Reply:
> This will tell me if I need to prioritize that.

Sounds good 👍